device/tt_soc_descriptor.cpp

```diff
@@ -168,13 +171,69 @@ void tt_SocDescriptor::load_core_descriptors_from_device_descriptor(YAML::Node &
     }
 }
 
+tt_xy_pair tt_SocDescriptor::calculate_grid_size(const std::vector<tt_xy_pair> &cores) {
+    if (cores.empty()) {
+        return {0, 0};
+    }
+
+    size_t x = 0;
```

Comment:
Would this be the same implementation as the following? I find it more readable:
```
std::unordered_set<size_t> x;
std::unordered_set<size_t> y;
for (auto core : cores) {
  x.insert(core.x];
  y.insert[core.y];
}
return {x.size(), y.size()};
```